### DISCUSSION TITLE
Loading smooth font on Rpi from example don't work

### DISCUSSION THREAD

**Discussion by rradomir** [Bodmer/TFT_eSPI#2443] [Q&A - Fonts]
RPi use own SD class. Maybe this causes problems with provided example esp32_smooth_font_sd. I deleted most SD-specific tools, like displaying size or type of card.
But font still can't be loaded due to:

`no matching function for call to 'TFT_eSPI::loadFont(arduino::String&, SDClass&)'`
This is line where error is:
`tft.loadFont(fileName, SD);`

I of course edited SD initialisation to match my hardware. Maybe I'm doing something wrong?

**Comment by Bodmer** [comment]
This is not something I have tried. The example was for the ESP32 and it looks like it is not compatible with the RP2040.  I would try to get the SD card examples working with the RP2040 first then integrate into the sketch. The SD and TFT cannot share the same SPI bus if the PIO interface option is used.

**Comment by rradomir** [comment]
I think PIO isn't used. 
I have RPi, [This](https://www.waveshare.com/wiki/Pico-ResTouch-LCD-3.5) Waveshare screen with sd card reader, I'm using this:
`SD.begin(SD_CS, tft.getSPIinstance())`
TFT and reader share all SPI pins, of course exlcuding CS pins, which are 9 for screen, and 22 for card reader. 
And all other functions and examples working good, for example loading JPG files from SDcard and viewing them on screen.
